src/stavenote.ts

```diff
@@ -1113,24 +1109,18 @@ export class StaveNote extends StemmableNote {
             yBottom - noteStemHeight + this.flag.getTextMetrics().actualBoundingBoxAscent;
 
       // Draw the Flag
-      ctx.save();
-      this.applyStyle(ctx, this.flagStyle);
-      this.flag.renderText(ctx, flagX, flagY);
-      ctx.restore();
+      this.flag.draw = (): void => {
+        this.flag.renderText(ctx, flagX, flagY);
+      };
+      this.flag.setContext(ctx).drawWithStyle();
```

Comment:
Interesting that we define this draw function on the fly.

Should the flag be its own class?

Reply:
done

Reply:
done